File: adapters/ruby/i18nt.rb
module I18nt
  class I18n
    def initialize(data)
      @data = data
    end

    def method_missing(name, *args, &block)
      key = name.to_sym
      if @data.key?(key)
        value = @data[key]
        if value.is_a?(Hash)
          I18n.new(value)
        else
          # 如果有参数，进行插值
          params = args.first
          if params.is_a?(Hash)
            result = value.to_s
            params.each { |k, v| result.gsub!("{#{k}}", v.to_s) }
            result
          else
            value
          end
        end
      else
        "[#{name}]"
      end
    end

    def respond_to_missing?(name, include_private = false)
      @data.key?(name.to_sym) || super
    end

    def to_s
      @data.is_a?(Hash) ? "" : @data.to_s
    end
  end
end

File: adapters/lua/i18nt.lua
local I18nt = {}
I18nt.__index = function(self, key)
    local val = rawget(self, "_data")[key]
    if type(val) == "table" then
        return setmetatable({ _data = val }, I18nt)
    end
    return val or ("[" .. key .. "]")
end

-- 支持调用进行插值：t.welcome({ name = "Antigravity" })
I18nt.__call = function(self, params)
    local text = rawget(self, "_data")
    if type(text) ~= "string" then return tostring(text) end
    
    if params then
        for k, v in pairs(params) do
            text = text:gsub("{" .. k .. "}", tostring(v))
        end
    end
    return text
end

function I18nt.new(data)
    return setmetatable({ _data = data }, I18nt)
end

return I18nt
